uint8 pin
uint32 us
---
uint8 result
